src/java/com/phenix/pct/PCT.java

```diff
@@ -627,82 +627,35 @@ protected static final int nextRandomInt() {
-            throw new IOException(message);
-        }
-    }
-
-    /**
-     * Cleans a directory without deleting it.
-     * 
-     * @param directory directory to clean
-     * @throws IOException in case cleaning is unsuccessful
-     */
-    private static void cleanDirectory(File directory) throws IOException {
         if (!directory.exists()) {
-            String message = directory + " does not exist";
-            throw new IllegalArgumentException(message);
-        }
-
-        if (!directory.isDirectory()) {
-            String message = directory + " is not a directory";
-            throw new IllegalArgumentException(message);
+            return;
         }
 
-        File[] files = directory.listFiles();
-        if (files == null) { // null if security restricted
-            throw new IOException("Failed to list contents of " + directory);
-        }
-
-        IOException exception = null;
-        for (int zz = 0; zz < files.length; zz++) {
-            try {
-                forceDelete(files[zz]);
-            } catch (IOException ioe) {
-                exception = ioe;
-            }
+        for (File f : directory.listFiles()) {
+            deleteFile(f);
         }
+        try {
+            Files.deleteIfExists(directory.toPath());
+        } catch (IOException caught) {
+            log(MessageFormat.format(Messages.getString("PCTRun.5"), directory.getAbsolutePath()),
+                    Project.MSG_VERBOSE);
 
-        if (null != exception) {
-            throw exception;
         }
     }
 
-    /**
-     * Deletes a file. If file is a directory, delete it and all sub-directories.
-     * <p>
-     * The difference between File.delete() and this method are:
-     * <ul>
-     * <li>A directory to be deleted does not have to be empty.</li>
-     * <li>You get exceptions when a file or directory cannot be deleted. (java.io.File methods
-     * returns a boolean)</li>
-     * </ul>
-     * 
-     * @param file file or directory to delete, must not be <code>null</code>
-     * @throws NullPointerException if the directory is <code>null</code>
-     * @throws FileNotFoundException if the file was not found
-     * @throws IOException in case deletion is unsuccessful
-     */
-    private static void forceDelete(File file) throws IOException {
-        if (file.isDirectory()) {
+    protected void deleteFile(File file) {
+        if (file == null)
+            return;
+        if (file.isDirectory())
             deleteDirectory(file);
-        } else {
-            boolean filePresent = file.exists();
-            if (!file.delete()) {
-                if (!filePresent) {
-                    throw new FileNotFoundException("File does not exist: " + file);
-                }
-                String message = "Unable to delete file: " + file;
-                throw new IOException(message);
-            }
+        try {
+            Files.deleteIfExists(file.toPath());
+        } catch (IOException caught) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)